File: Quiz/DB/fix_modules.py
import os

files_to_fix = [
    'MD2-120 câu_updated.csv',
    'MD5-120 câu_updated.csv'
]
target_lines = 121  # Header + 120 questions

for file_name in files_to_fix:
    file_path = file_name
    backup_path = file_name + '.bak'
    
    try:
        if not os.path.exists(file_path):
            print(f"File not found: {file_path}")
            continue

        # Read lines
        with open(file_path, 'r', encoding='utf-8') as f:
            lines = f.readlines()

        print(f"Processing {file_name}...")
        print(f"Original line count: {len(lines)}")

        if len(lines) <= target_lines:
            print(f"{file_name} is already within the target line count.")
        else:
            # Backup original
            with open(backup_path, 'w', encoding='utf-8') as f:
                f.writelines(lines)
            print(f"Backup created at {backup_path}")

            # Truncate
            new_lines = lines[:target_lines]
            
            # Write back
            with open(file_path, 'w', encoding='utf-8') as f:
                f.writelines(new_lines)
            print(f"{file_name} truncated to {len(new_lines)} lines.")

    except Exception as e:
        print(f"Error processing {file_name}: {e}")

File: Module_1-6/DB/fix_modules.js
const fs = require('fs');
const path = require('path');

const filesToFix = [
    'MD2-120 câu_updated.csv',
    'MD5-120 câu_updated.csv'
];
const targetLines = 121; // Header + 120 questions

filesToFix.forEach(fileName => {
    const filePath = path.join(__dirname, fileName);

    try {
        if (!fs.existsSync(filePath)) {
            console.log(`${fileName}: Not Found`);
            return;
        }

        const content = fs.readFileSync(filePath, 'utf8');
        // Split by newline, handling both \r\n and \n
        const lines = content.split(/\r?\n/);

        // Filter out empty lines at the end if any, to get accurate count
        const nonEmptyLines = lines.filter(line => line.trim() !== '');

        console.log(`${fileName} Original Lines: ${lines.length} (Non-empty: ${nonEmptyLines.length})`);

        if (lines.length > targetLines) {
            // Truncate
            const newLines = lines.slice(0, targetLines);
            const newContent = newLines.join('\n');

            fs.writeFileSync(filePath, newContent, 'utf8');
            console.log(`${fileName} Truncated to: ${newLines.length}`);
        } else {
            console.log(`${fileName} No change needed.`);
        }

    } catch (err) {
        console.error(`Error processing ${fileName}:`, err);
    }
});
